go/gas-oracle/gasprices/gas_price_updater.go

```diff
@@ -63,14 +62,23 @@ func (g *GasPriceUpdater) UpdateGasPrice() error {
 	if err != nil {
 		return err
 	}
-	if latestBlockNumber < uint64(g.epochStartBlockNumber) {
+	if latestBlockNumber < g.epochStartBlockNumber {
 		return errors.New("Latest block number less than the last epoch's block number")
 	}
+
+	// Accumulate the amount of gas that has been used in the epoch
+	totalGasUsed := uint64(0)
+	for i := g.epochStartBlockNumber; i < latestBlockNumber; i++ {
+		gasUsed, err := g.getGasUsedByBlockFn(new(big.Int).SetUint64(i))
+		if err != nil {
+			return err
+		}
+		totalGasUsed += gasUsed
+	}
+
 	averageGasPerSecond := GetAverageGasPerSecond(
-		g.epochStartBlockNumber,
-		latestBlockNumber,
+		totalGasUsed,
```

Comment:
Style nit: I would personally suggest either:
1. Removing the `GetAverageGasPerSecond(...)` function and replace this line with simply: `averageGasPerSecond := float64(gasUsed / epochLengthSeconds)`; or
1. Moving the total gas calculation logic into `GetAverageGasPerSecond(...)`.

Just seems like right now this is an awkward middle ground where we have a function for some super simple / basic logic (calculating an average).

Reply:
I originally left it because there was a unit test for it, but you are right its pretty simple. I removed it along with its unit test